### PR TITLE
fixed: generate email urls cloud functions side

### PR DESCRIPTION
**Pull request by yspringfield**
Fixed: #7616

### CODE REVIEW

File: apps/backend-functions/src/notification.ts [hunk 1]
```diff
@@ -678,9 +678,12 @@ async function sendContractStatusChangedConfirmation(recipient: User, notificati
     contract, title, app, isRecipientBuyer, toUser, recipientOrg, counterOfferSenderOrg
   } = await getNegotiationUpdatedEmailData(recipient, notification);
 
+  let pageURL = `${appUrl.content}/c/o/dashboard/sales/${contract.id}/view`;
+  if(isRecipientBuyer) pageURL = `${appUrl.content}/c/o/marketplace/offer/${contract.offerId}/${contract.id}`;
```
Comment (RemcoSimonides) [inline apps/backend-functions/src/notification.ts:682]:
```suggestion
  const pageURL = isRecipientBuyer 
    ?  `${appUrl.content}/c/o/dashboard/sales/${contract.id}/view`
    :  `${appUrl.content}/c/o/marketplace/offer/${contract.offerId}/${contract.id}`;
```

File: apps/backend-functions/src/notification.ts [hunk 2]
```diff
@@ -678,9 +678,12 @@ async function sendContractStatusChangedConfirmation(recipient: User, notificati
...
+
   const data = {
-    user: toUser, baseUrl: appUrl.content, org: recipientOrg,
-    contract, title, isRecipientBuyer
+    user: toUser, org: recipientOrg,
+    contract, title, pageURL, app:{name:appName.catalog}
```
Comment (RemcoSimonides) [inline apps/backend-functions/src/notification.ts:686]: Give the object some air (a.k.a. space) ^^
```suggestion
  const data = {
    user: toUser,
    org: recipientOrg,
    contract,
    title,
    pageURL,
    app: { name: appName.catalog }
  };
```

File: apps/backend-functions/src/templates/mail.ts [hunk 3]
```diff
@@ -368,17 +368,23 @@ export function adminOfferCreatedConfirmationEmail(toUser: UserEmailData, org: O
 /**To inform buyer that his offer has been successfully created. */
 export function buyerOfferCreatedConfirmationEmail(toUser: UserEmailData, org: OrganizationDocument, offerId: string, bucket: Bucket<Timestamp>): EmailTemplateRequest {
   const contracts = bucket.contracts.map(contract => createMailContract(contract));
-  const data = { org, bucket: { ...bucket, contracts }, user: toUser, baseUrl: appUrl.content, offerId };
+  const pageURL = `${appUrl.content}/c/o/marketplace/offer/${offerId}`;
+  const data = {
+    app: { name: appName.catalog }, bucket: { ...bucket, contracts },
+    user: toUser, pageURL, offerId, org
+  };
```
Comment (RemcoSimonides) [inline apps/backend-functions/src/templates/mail.ts:375]: Same here, just put all the properties on a new line. It's easier to read.

File: apps/backend-functions/src/templates/mail.ts [hunk 4]
```diff
@@ -368,17 +368,23 @@ export function adminOfferCreatedConfirmationEmail(toUser: UserEmailData, org: O
...
   return { to: toUser.email, templateId: templateIds.offer.toBuyer, data };
 }
 
 export function counterOfferRecipientEmail(
   toUser: UserEmailData, senderOrg: OrganizationDocument, offerId: string,
   title: MovieDocument, contractId: string, options: { isMailRecipientBuyer: boolean }
 ): EmailTemplateRequest {
+  let pageURL = `${appUrl.content}/c/o/dashboard/sales/${contractId}/view`;
+  if (options.isMailRecipientBuyer) pageURL = `${appUrl.content}/c/o/marketplace/offer/${offerId}/${contractId}`;
```
Comment (RemcoSimonides) [inline apps/backend-functions/src/templates/mail.ts:384]: Use `const pageURL = x ? y : z` here too

File: apps/backend-functions/src/templates/mail.ts [hunk 5]
```diff
@@ -368,17 +368,23 @@ export function adminOfferCreatedConfirmationEmail(toUser: UserEmailData, org: O
...
   const data = {
-    user: toUser, baseUrl: appUrl.content, offerId, org: senderOrg,
-    contractId, title, isRecipientBuyer: options.isMailRecipientBuyer
+    user: toUser, offerId, org: senderOrg, pageURL,
+    title, app: { name: appName.catalog }
```
Comment (RemcoSimonides) [inline apps/backend-functions/src/templates/mail.ts:387]: new  lines

File: apps/backend-functions/src/templates/mail.ts [hunk 6]
```diff
@@ -389,9 +395,11 @@ export function counterOfferSenderEmail(
 ): EmailTemplateRequest {
   const terms = createMailTerm(negotiation.terms);
   const currency = staticModel['movieCurrencies'][negotiation.currency];
+  let pageURL = `${appUrl.content}/c/o/dashboard/sales/${contractId}/view`;
+  if (options.isMailRecipientBuyer) pageURL = `${appUrl.content}/c/o/marketplace/offer/${offerId}/${contractId}`;
```
Comment (RemcoSimonides) [inline apps/backend-functions/src/templates/mail.ts:399]: `x ? y : z`

File: apps/backend-functions/src/templates/mail.ts [hunk 7]
```diff
@@ -389,9 +395,11 @@ export function counterOfferSenderEmail(
...
   const data = {
-    user: toUser, baseUrl: appUrl.content, offerId, org,
-    contractId, isRecipientBuyer: options.isMailRecipientBuyer,
+    user: toUser, pageURL, offerId, org,
+    contractId, app: { name: appName.catalog },
     negotiation: { ...negotiation, terms, currency },
     title: title.title.international
   };
```
Comment (RemcoSimonides) [inline apps/backend-functions/src/templates/mail.ts:405]: new lines

File: apps/backend-functions/src/templates/mail.ts [hunk 8]
```diff
@@ -405,9 +413,10 @@ export function offerAcceptedOrDeclined(
 ): EmailTemplateRequest {
   const isOfferAccepted = offer.status === 'accepted';
   const acceptedContracts = contracts.filter(contract => contract.status === 'accepted');
+  const pageURL = `${appUrl.content}/c/o/marketplace/offer/${offer.id}`;
   const data = {
     contracts: isOfferAccepted ? acceptedContracts : contracts,
-    baseUrl: appUrl.content, offer, user
+    offer, user, pageURL, app: { name: appName.catalog }
   };
```
Comment (RemcoSimonides) [inline apps/backend-functions/src/templates/mail.ts:420]: new lines

**Review by RemcoSimonides — APPROVED** [review]

**Review by GrandSchtroumpf — APPROVED** [review]

**Pull request merged**